plugins/modules/win_psmodule_info.py

```diff
@@ -0,0 +1,439 @@
+      type: bool
+    module_list:
+      description:
+        - A list of modules packaged with this module.
+        - This value is not often returned and the modules are not automatically processed.
+      type: list
+      elements: dict
+      contains:
+        name:
+          description:
+            - The name of the module.
+            - This may also be a path to the module file.
+          type: str
+          sample: '.\WindowsUpdateLog.psm1'
+        guid:
+          description:
+            - The GUID of the module.
+          type: str
+          sample: 82fdb72c-ecc5-4dfd-b9d5-83cf6eb9067f
+        version:
+          description:
+            - The minimum version of the module.
+          type: str
+          sample: '2.0'
+        maximum_version:
+          description:
+            - The maximum version of the module.
+          type: str
+          sample: '2.9'
+        required_version:
+          description:
+            - The exact version of the module required.
+          type: str
+          sample: '3.1.4'
+    nested_modules:
+      description:
+        - A list of modules nested with and loaded into the scope of this module.
+        - This list contains full module objects, so each item can have all of the properties listed here, including C(nested_modules).
+      type: list
+      elements: dict
+    required_modules:
+      description:
+        - A list of modules required by this module.
+        - This list contains full module objects, so each item can have all of the properties listed here, including C(required_modules).
+        - These module objects may not contain full information however, so you may see different results than if you had directly queried the module.
+      type: list
+      elements: dict
+    required_assemblies:
+      description:
+        - A list of assemblies that the module requires.
+        - The values may be a simple name or a full path.
+      type: str
+      sample:
+        - Microsoft.Management.Infrastructure.CimCmdlets.dll
+        - Microsoft.Management.Infrastructure.Dll
+    package_management_provider:
+      description:
+        - If the module was installed from PowerShellGet, this is the package management provider used.
+      type: str
+      sample: NuGet
+    power_shell_host_name:
+      description:
+        - The name of the PowerShell host that the module requires.
+      type: str
+      sample: Windows PowerShell ISE Host
+    power_shell_host_version:
+      description:
+        - The version of the PowerShell host that the module requires.
+      type: str
+      sample: '1.1'
+    power_shell_version:
+      description:
+        - The minimum version of PowerShell that the module requires.
+      type: str
+      sample: '5.1'
+    prefix:
+      description:
+        - The default prefix applied to C(Verb-Noun) commands exported from the module, resulting in C(Verb-PrefixNoun) names.
+      type: str
+    private_data:
```

Comment:
I'm not sure if this is actually what the value should be but `private_data` in some of the nested modules of `NetAdapter` are showing up as

```
private_data:
  CmdletOverObjects:
    ClassName: ROOT/StandardCimv2/MSFT_....
    CmdletAdpater: Microsoft.PowerShell.Cmdletization.Cim.CimCmdletAdapter
```

I think is is correct but probably just something to look over to make sure.

Reply:
Yeah that's expected. I actually handle `private_data` specially specifically for non-Script/Manifest modules because of this. From the comment on lines 93-97:

> PrivateData can contain any data but a module manifest is a static file that can't contain references or problematic types like [Type]. Unfortunately some module types like CIM (and presumably binary?) seem to be able to populate that with whatever they want.
> 
> As a precaution then, for module type that is not Script or Manifest, we limit PrivateData to a shallow depth.

It was those nested CIM modules that brought it to my attention; they had `[Type]` values which balloon serialization really quickly :(

Reply:
Thanks for confirming